docs(error/areq): describe your change...

nikhilknoldus opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**docs** Document Update.

**What is the current behavior? (You can also link to an open issue here)**

Application stops, when this error occur we get official [link](https://docs.angularjs.org/error/ng/areq?p0=SourceCtrl&p1=not) but this does not solve the error.

**What is the new behavior (if this is a feature change)?**
## 

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Encountered this error many times, the provided details are very encrypted (don't seem practical), i have added a line that is 99% practical and main cause for this error. These 3 points that i have added are seriously core reasons for this error.

Commit:
docs(error/areq): describe your change...

Encountered this error many times, the provided details are very encrypted (don't seem practical), i have added a line that is 99% practical and main cause for this error. These 3 points that i have added are seriously core reasons for this error.

nikhilknoldus commented:
I signed it.

nikhilknoldus commented:
CI failed due to your server timeout, not mine. Please review.

Narretz commented:
The error is too specific. `areq` is not only thrown for controllers, but for other parts of the code, too.

nikhilknoldus commented:
@Narretz your view is right but the doc's lines are not worth at all, written so complicatedly, number of angular developers i know who open this bug line from browser console, read it and say what's this? They don't infer anything from this detail. The comment which i have written matching major error criteria and the earlier doesn't.

Narretz commented:
But your change only mentions controller, which is is also confusing. If someone gets the error without having ngController, then this is not helpful at all.
I'm okay with including some note about syntax errors, but it should be less specific. If you have example code that shows a common example how this error can happen, then we can include it.

Commit:
Update areq.ngdoc

Earlier commit was focusing on controller only, but current commit is covering large area of this error, that is mostly the reason.

nikhilknoldus commented:
@Narretz Martin your view is again right in this case, but as i mentioned, syntax error and earlier error doesn't infer it, and yes my commit is focusing on controller as you said, i can modify it for larger criteria. And for the example thing: my project is quite big (for an international bank), i do have lots of controllers too and in that case watching this error horribly blank our(js developers) mind when something like this happens: Examples:
1-    angular.module('someApp').controller(-'SomeCntrl', SomeCtrl);
*notice the hyphen before. it may be any symbol, char etc. That happens by mistake obviously.

and this is only 1 scenario there are much that we face alot.

I have updated my commit as well.

Thanks !!

nikhilknoldus commented:
Hi @Narretz any update regarding same.

Narretz commented:
I've taken another look and I think you are right that this error is confusing when it is thrown with for controllers. I've opened a PR that throws a better error here: https://github.com/angular/angular.js/pull/15015

nikhilknoldus commented:
Hey thanks Martin for considering my PR.
